packages/flutter/lib/src/material/material.dart

```diff
@@ -656,5 +656,43 @@ class _InkHighlight extends InkFeature implements InkHighlight {
       _paintHighlight(canvas, rect.shift(originOffset), paint);
     }
   }
+}
+
+/// A widget that automatically dispatches a [LayoutChangedNotification] when
+/// the layout of its child changes.
+///
+/// Useful especially when having some complex, layout-changing animation within
+/// [Material] that is also interactive.
+class LayoutChangedNotifier extends SingleChildRenderObjectWidget {
+  /// Creates a [LayoutChangedNotifier] that dispatches layout changed
+  /// notifications when [child] changes layout.
+  LayoutChangedNotifier({
+    Key key,
+    Widget child
+  }) : super(key: key, child: child);
+
+  @override
+  _RenderLayoutChangedWithCallback createRenderObject(BuildContext context) {
+    return new _RenderLayoutChangedWithCallback(
+      onLayoutChangedCallback: () {
+        new LayoutChangedNotification().dispatch(context);
+      }
+    );
+  }
+}
+
+class _RenderLayoutChangedWithCallback extends RenderProxyBox {
+  _RenderLayoutChangedWithCallback({
+    RenderBox child,
+    this.onLayoutChangedCallback
+  }) : super(child);
 
+  VoidCallback onLayoutChangedCallback;
+
+  @override
+  void performLayout() {
+    if (onLayoutChangedCallback != null)
```

Comment:
This doesn't work. There's no way to guarantee that you'll be called whenever the child changes layout. For example, this won't work if the child is just transforming the child (RenderTransform), or if the children are being moved without the constraints changing, etc.